course_info/tests/jasmine/controllers/DetailsControllerSpec.js

```diff
@@ -559,4 +566,161 @@ describe('Unit testing DetailsController', function () {
             expect(result).toEqual('ABC');
         });
     });
+
+    describe('associateNewSiteHandleKey', function(){
+        it('should call associateNewSite when a user hits the enter key', function(){
+            spyOn(dc, 'validNewSiteURL').and.returnValue(true);
+            spyOn(dc, 'associateNewSite');
+            var aEvent = angular.element.Event('keydown');
+            aEvent.keyCode = 13;
+            $("input").trigger(aEvent);
+            dc.associateNewSiteHandleKey(aEvent);
+            expect(dc.validNewSiteURL).toHaveBeenCalled();
+            expect(dc.associateNewSite).toHaveBeenCalled();
+
+        });
+
+        it('should not call associateNewSite when a user enteres an invalid url', function(){
+            spyOn(dc, 'validNewSiteURL').and.returnValue(false);
+            spyOn(dc, 'associateNewSite');
```

Comment:
I guess a few of these lines could be `beforeEach()` and `afterEach()` for DRY-ness, but that may be overkill - with only two tests, it does read clearly this way.